repair/repair_task.hh

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "task_manager/task_manager.hh"
+
+class repair_task : public task_manager::task::impl {
+public:
+    repair_task(task_manager::module_ptr module, task_id id, unsigned sequence_number = 0, std::string keyspace = "", std::string table = "", std::string type = "", std::string entity = "", task_manager::foreign_task_ptr&& parent = nullptr) 
+        : task_manager::task::impl(module, id, sequence_number, std::move(keyspace), std::move(table), std::move(type), std::move(entity), std::move(parent)) {
+        _status.progress_units = "SSTables";
+    }
+
+    future<task_manager::task::progress> get_progress() const override {
+        if (_children.empty()) {
+            return make_ready_future<task_manager::task::progress>(_progress);
+        }
+
+        return map_reduce(_children, [] (auto& child) {
+            return smp::submit_to(child.get_owner_shard(), [&child] {
+                return child->get_progress();
+            });
+        }, task_manager::task::progress{}, [] (task_manager::task::progress l, task_manager::task::progress r) {
+            task_manager::task::progress res{};
+            res.total = l.total + r.total;
+            res.completed = l.completed + r.completed;
+            return res;
+        });
+    }
+
+    ::is_abortable is_abortable() const noexcept override {
```

Comment:
Currently, all repair tasks are not abortable.
This too can be the default implementation of the virtual method.